src/RBF/rbf.hpp

```diff
@@ -74,12 +98,15 @@ class RBFKernel{
 
     std::vector<double>                 m_error;    /**<Interpolation error of a field evaluated on each RBF node (auxiliary memeber used in Greedy algorithm).*/
 
-    protected:
+protected:
     std::vector<std::vector<double>>    m_value;    /**< displ value to be interpolated on RBF nodes */
     std::vector<std::vector<double>>    m_weight;   /**< weight of your RBF interpolation */
     std::vector<bool>                   m_activeNodes;   /**<Vector of active/inactive node (m_activeNodes[i] = true/false -> the i-th node is used/not used during RBF evaluation).*/   
     int m_maxFields;                                /**< fix the maximum number of fields that can be added to your class*/
     int m_nodes;                                    /**<Number of RBF nodes.*/
+    std::set<int> m_polyActiveBasis;                /**< Active terms of linear polynomial, -1 is constant, 0,1,2 the system coordinates */
```

Comment:
In case of skipping checks on coordinates already set as in the current implementation yes, it can be a vector. 
In case of loop always from 0 to 2 as proposed in https://github.com/optimad/bitpit/pull/434#discussion_r1417048034, I think that has to remain a set to avoid multiple insert of the same coordinate index.